tex/generic/pgf/utilities/pgfkeys.code.tex

```diff
@@ -379,7 +380,9 @@
 \def\pgfkeys@case@one{%
   \pgfkeysifdefined{\pgfkeyscurrentkey/.@cmd}%
   {\pgfkeysgetvalue{\pgfkeyscurrentkey/.@cmd}{\pgfkeys@code}%
-   \expandafter\pgfkeys@code\pgfkeyscurrentvalue\pgfeov}
+    \ifx\pgfkeys@code\relax\expandafter\pgfkeys@firstoftwo\else\expandafter\pgfkeys@secondoftwo\fi
+    {\pgfkeys@unknown}%
+    {\expandafter\pgfkeys@code\pgfkeyscurrentvalue\pgfeov}}
   {\pgfkeys@case@two}%
 }
```

Comment:
With the change to `\pgfkeysifdefined` in https://github.com/pgf-tikz/pgf/pull/1132/commits/b8492921a47dba146c9bd1aaa2946082c1c37935, this newly added logic is unnecessary.

I know it's required to pass the current test 1, which is equivalent to
```tex
\pgfkeyslet{/foo/.@cmd}{\relax}
\pgfkeys{/foo}
```
but as I said in https://github.com/pgf-tikz/pgf/issues/1131#issuecomment-1025153232, in normal usage, there's no way for a `/.@cmd` subkey to be equal to `\relax` (as far as `\pgfkeysgetvalue` and the like are now `\csname` safe). One could wrongly write `\pgfkeyslet{/foo/.@cmd}{<token>}` in which the arg-spec of `<token>` doesn't end with `\pgfeov` and get similar errors.

This added logic `\ifx\pgfkeys@code\relax...` only save the case when `<token> == \relax`.

Reply:
Since the aim was never to catch any wrong token, but just an erroneous let to `\relax` I think this could be fine. But you got a point with the situation now being that through official interfaces it shouldn't be possible to end up with such an accidental assignment. So I have no strong opinion on this (other than that there could be code outside the scope of "our" control that could do such assignments by unconditionally using `\csname`).

So the team should decide here: What is in the scope of these tests? Should the file be self-secure or also catch the most simple other error source. I guess everyone agrees on that the purpose shouldn't be detection of *every* error case of wrong assignment.

Reply:
It makes for a much nicer error and an accidental `\relax` is easy to creep in. There are quite a few packages in TeX Live that write `\pgfk@...` directly.

Reply:
> There are quite a few packages in TeX Live that access `\pgfk@...` directly.

So this PR actually introduces a breaking change (that a not-so-right behavior since 2007 is now changed) which might affect more than I though before? I'd suggest to move the changelog entry under section `### BREAKING CHANGES`.